### PR TITLE
feat(calm-hub): Add Timeline Tools and Bug Fix Architecture Version Create

### LINKED ISSUE #2522: MCP architecture-version + missing timeline tools
### Steps to Reproduce:
 1. Start calm-hub with default config (`allow.put.operations=false`).
 2. From an MCP client call the `updateArchitecture` tool (described as "Publish a new version of an existing architecture") with a new semver.
 3. From an MCP client try to list/get/create a timeline.

### Expected Result:
 - Publishing a new architecture version should succeed — it is a POST/append, not a PUT — matching `PatternTools.createPatternVersion` and the REST `POST /namespaces/{ns}/architectures/{id}/versions/{v}` endpoint.
 - Timeline MCP tools should exist (list / list versions / get / create / create version), mirroring `TimelineResource`.

### Actual Result:
 - `updateArchitecture` is gated by `allowPutOperations` and internally calls `updateArchitectureForVersion` (PUT semantics) instead of `createArchitectureForVersion`. New versions cannot be added without enabling PUT.
 - No `TimelineTools` class exists; timelines are completely unreachable via MCP.

### Environment:
 calm-hub `main`, Quarkus 3.29.x, default profile.

### Additional Context:
 Proposed fix:
 - Split `ArchitectureTools.updateArchitecture` into `createArchitectureVersion` (POST, ungated, calls `createArchitectureForVersion`, handles `ArchitectureVersionExistsException`) and a separate `updateArchitecture` (PUT, gated by `allowPutOperations`) — same shape as `PatternTools`.
 - Add `TimelineTools` exposing list/listVersions/get/create/createVersion against `TimelineStore`, with unit tests mirroring `TestPatternToolsShould`.

### PR DESCRIPTION
**Pull request by jpgough-ms**
…reation

- Add TimelineTools with 6 MCP tools: listTimelines, listTimelineVersions, getTimeline, createTimeline, createTimelineVersion, updateTimeline
- Fix ArchitectureTools: split updateArchitecture into separate createArchitectureVersion (POST, ungated) and updateArchitecture (PUT, gated by allowPutOperations) to match the correct HTTP semantics
- Add unit tests for TimelineTools covering all tools and edge cases, including regression guard that createTimelineVersion works when allowPutOperations=false
- Extend MongoMcpIntegration and NitriteMcpIntegration with timeline end-to-end tests using isolated namespaces (mcp-integration/mcp-nitrite)
- Fix MongoTimelineIntegration and NitriteTimelineIntegration to capture the created timeline ID dynamically from the Location header instead of hardcoding ID 1, resolving a pre-existing test ordering fragility

Fixes #2522

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🎨 Code style/formatting changes
- [ ] ♻️ Refactoring (no functional changes)
- [ ] ⚡ Performance improvements
- [ ] ✅ Test additions or updates
- [ ] 🔧 Chore (maintenance, dependencies, CI, etc.)

## Affected Components

- [ ] CLI (`cli/`)
- [ ] Schema (`calm/`)
- [ ] CALM AI (`calm-ai/`)
- [ ] CALM Hub (`calm-hub/`)
- [ ] CALM Hub UI (`calm-hub-ui/`)
- [ ] CALM Server (`calm-server/`)
- [ ] CALM Widgets (`calm-widgets/`)
- [ ] Documentation (`docs/`)
- [ ] Shared (`shared/`)
- [ ] VS Code Extension (`calm-plugins/vscode/`)
- [ ] Dependencies
- [ ] CI/CD

## Testing

- [ ] I have tested my changes locally
- [ ] I have added/updated unit tests
- [ ] All existing tests pass

## Checklist
- [ ] My commits follow the [conventional commit format](https://www.conventionalcommits.org/)
- [ ] I have updated documentation if necessary
- [ ] I have added tests for my changes (if applicable)
- [ ] My changes follow the project's coding standards

### CODE REVIEW

File: calm-hub/src/integration-test/java/integration/MongoTimelineIntegration.java [hunk 1]
```diff
@@ -74,20 +76,29 @@ void end_to_end_create_a_timeline() {
...
-                .header("Location", containsString("calm/namespaces/finos/timelines/1"));
+                .header("Location", containsString("calm/namespaces/finos/timelines/"))
+                .extract().header("Location");
+
+        int timelinesIdx = location.indexOf("/timelines/") + "/timelines/".length();
+        String idStr = location.substring(timelinesIdx);
+        if (idStr.contains("/")) {
+            idStr = idStr.substring(0, idStr.indexOf('/'));
+        }
+        createdTimelineId = Integer.parseInt(idStr);
```
Comment (rocketstack-matt) [inline calm-hub/src/integration-test/java/integration/MongoTimelineIntegration.java]: Manual substring parsing of the `Location` header is brittle and harder to read than necessary. The MCP integration tests already use a compiled `ID_PATTERN` regex for the same job (see `MongoMcpIntegration`). Suggest extracting a small helper (e.g. `extractIdFromLocation(String location, String resource)`) — or just using `Pattern.compile("/timelines/(\\d+)")` inline — and reusing it here and in `NitriteTimelineIntegration`. Non-blocking; removing the hardcoded `1` is the important part.

File: calm-hub/src/integration-test/java/integration/NitriteTimelineIntegration.java [hunk 2]
```diff
@@ -41,20 +43,28 @@ void end_to_end_create_a_timeline() {
...
-                .header("Location", containsString("calm/namespaces/finos/timelines/1"));
+                .header("Location", containsString("calm/namespaces/finos/timelines/"))
+                .extract().header("Location");
+
+        int timelinesIdx = location.indexOf("/timelines/") + "/timelines/".length();
+        String idStr = location.substring(timelinesIdx);
+        if (idStr.contains("/")) {
+            idStr = idStr.substring(0, idStr.indexOf('/'));
+        }
+        createdTimelineId = Integer.parseInt(idStr);
```
Comment (rocketstack-matt) [inline calm-hub/src/integration-test/java/integration/NitriteTimelineIntegration.java]: Same comment as on `MongoTimelineIntegration`: duplicated substring parsing logic. A shared helper using a regex would be cleaner and match the `ID_PATTERN` idiom already used by the MCP integration tests. Non-blocking.

File: calm-hub/src/main/java/org/finos/calm/mcp/tools/TimelineTools.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        } catch (TimelineNotFoundException e) {
+            logger.warn("Timeline [{}] not found in namespace [{}]", timelineId, namespace, e);
+            return McpResponseFormatter.entityNotFound("timeline", timelineId, namespace);
+        } catch (TimelineVersionExistsException e) {
+            logger.warn("Version [{}] already exists for timeline [{}] in namespace [{}]", version, timelineId, namespace, e);
+            return ToolResponse.error("Error: Version '" + version + "' already exists for timeline " + timelineId + ".");
+        }
+    }
+
+    @Tool(description = "Update the content of an existing timeline version. Requires PUT operations to be enabled on this CalmHub instance.")
```
Comment (rocketstack-matt) [inline calm-hub/src/main/java/org/finos/calm/mcp/tools/TimelineTools.java:186]: Nice that the description explicitly calls out the PUT-operations gate — useful for MCP clients trying to discover why a tool errors. Worth noting that `PatternTools.updatePattern` does **not** carry the equivalent caveat in its description, even though it has the same `checkMutationAllowed(allowPutOperations)` gate. Suggest a small follow-up to align `PatternTools.updatePattern`'s `@Tool` description so all three update tools advertise the gate consistently. (Don't drop it here — the new wording is the correct one to standardise on.)

File: calm-hub/src/test/java/org/finos/calm/mcp/tools/TestTimelineToolsShould.java [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+        String huge = "{\"x\":\"" + "a".repeat(100_001) + "\"}";
+
+        ToolResponse result = timelineTools.createTimelineVersion("workshop", 1, "1.1.0", huge);
+
+        assertThat(result.isError(), is(true));
+        assertThat(text(result), containsString("Timeline JSON"));
+        verifyNoInteractions(timelineStore);
+    }
+
+    // --- updateTimeline (PUT — requires allowPutOperations) ---
```
Comment (rocketstack-matt) [inline calm-hub/src/test/java/org/finos/calm/mcp/tools/TestTimelineToolsShould.java:412]: Minor: compared to the `createTimeline` / `createTimelineVersion` sections, the `updateTimeline` validation coverage is a bit thinner. Currently no direct assertions for:

- blank/invalid `namespace` (e.g. `bad ns`, null, empty)
- blank `timelineJson`
- non-positive `timelineId` (e.g. `0`, `-1`)

These branches are exercised transitively by the existing `reject_*_for_create_timeline` and `create_timeline_version_*` tests via the shared `McpValidationHelper`, so JaCoCo's 90% per-class gate is likely satisfied. But for parity with the create-side coverage — and to guard against future drift if `updateTimeline`'s validation chain ever diverges from the create tools' — it'd be worth adding the three small tests above. Non-blocking.

**Review by rocketstack-matt — COMMENTED** [review]
A few additional non-blocking observations to complement Copilot's review.

**Review by rocketstack-matt — APPROVED** [review]
